pkg/composer/serviceparser/serviceparser_test.go

```diff
@@ -174,8 +174,8 @@ volumes:
 	assert.Assert(t, in(wp1.RunArgs, "--log-driver=json-file"))
 	assert.Assert(t, in(wp1.RunArgs, "--log-opt=max-size=5K"))
 	assert.Assert(t, in(wp1.RunArgs, "--log-opt=max-file=2"))
-	assert.Assert(t, in(wp1.RunArgs, "--add-host=test.com:172.19.1.1"))
-	assert.Assert(t, in(wp1.RunArgs, "--add-host=test2.com:172.19.1.2"))
+	assert.Assert(t, in(wp1.RunArgs, "--add-host=test.com:[172.19.1.1]"))
+	assert.Assert(t, in(wp1.RunArgs, "--add-host=test2.com:[172.19.1.2]"))
```

Comment:
This seems to generate an invalid `/etc/hosts`

```
$ nerdctl run --rm --add-host=example.com:[192.168.12.34] busybox cat /etc/hosts
# <nerdctl>
127.0.0.1       localhost localhost.localdomain
::1             localhost localhost.localdomain
[192.168.12.34] example.com
10.4.0.45       d39e31c01c00 busybox-d39e3
# </nerdctl>
```

Reply:
Interesting.

Let me look into it.

Reply:
So, compose now return []string instead of string.

I have no idea why oO. Is the hope that you would get some form of cheapo load balancing?

Passing multiple times the same host with different ips is not going to work.